File: Front/lib/Types/Vacancy/Vacancy.ts
import {CompanyShortInfo} from "@/lib/Types/Companies/CompanyShortInfo";

export default class Vacancy {
    vacancyId: string;
    createdAt: Date;
    updatedAt?: Date;
    title: string;
    description?: string;
    salary?: string;
    specialization?: string;
    experience?: string;
    companyShortInfo? : CompanyShortInfo
    
    
    constructor(VacancyId: string, CreatedAt: Date, title: string, description?: string, salary?: string, specialization?: string, experience?: string) {
        this.vacancyId = VacancyId;
        this.createdAt = CreatedAt;
        this.title = title;
        this.description = description;
        this.salary = salary;
        this.specialization = specialization;
        this.experience = experience;
    }
}

//fix returning after delete application

File: Back/GlobalModels/Vacancy/Vacancy.cs
using System.ComponentModel.DataAnnotations;
using MongoDB.Entities;
using VacancyService.Models;

namespace GlobalModels.Vacancy;

public class Vacancy : Entity, IVacancyInputFields
{
    [Required]
    public Guid VacancyId { get; set; }
    [Required]
    public string Title { get; set; }
    [Required]
    public string Description { get; set; }
    
    public string Salary { get; set; }
    
    public string Experience { get; set; }
    
    public string Specialization { get; set; }
    
    public DateTime CreatedAt { get; set; } 
    
    public DateTime UpdatedAt { get; set; }
    [Required]
    public CompanyShortInfo CompanyInfo { get; set; }
    
    
}